whitepaper/dsnp_whitepaper.tex

```diff
@@ -227,69 +261,69 @@ \subsection{User Ownership and Delegation}\label{sec:user_ownership and_delegati
+direct data ownership, rights delegation, and identity verification.
+
+\subsubsection{Direct Data Ownership}
+
+Many blockchain-based projects that employ smart contracts use a single contract instance to
+store multiple user ownership rights. While this approach works for simple use cases, such
+as nonupgradable smart contracts associating ERC-20\cite{erc20} token values with unique
+addresses, storing multiple users' data in a single smart contract means someone other than
+each user has the right to upgrade the common contract. Whoever has upgrade rights
+ultimately controls the contract, and therefore the functionality of the contract and the
+data it stores.\footnote{The first edition of this paper argued for giving users choice in
+  how and when to upgrade their Social Identity as represented specifically by a blockchain
+  smart contract. Further research and development led to the conclusion that creating a
+  coherent public space would only be possible if the participants agreed to follow the same
+  version (with allowances for forward and backward compatibility) of the protocol. The text
+  has been updated to reinforce the important role that user opt-in still plays, but that it
+  may be implemented using a variety of approaches.} Instead, the protocol guarantees that
+each user's control key retains sole control of their personal data and social graph.
+
+One major ramification of direct ownership is that the user should always have agency in the
+adoption of new system logic (for example, smart contract upgrades or blockchain core logic
+upgrades). At minimum, users require the ability to opt out of their association with a
+given consensus system. Ideally, consensus systems will enable democratic participation in
+the governance and adoption of logic related to this protocol; the precise mechanisms that
+help achieve this (DAOs, co-ops, etc.) are an area of active research within the technology
+community.  This avoids the case of an outside party having unilateral rights to change the
+behavior of the system without the user's consent. An outside party in possession of
+unilateral access rights would be equivalent to the status quo, meaning that existing social
+network providers have the power to revoke user account access. For a Social Identity, no
+third party has this capability.
 
 Our \say{design by contract} approach provides another extension to ownership: complete
 implementation change. Although we expect most users to use the same implementation, other
 implementations may arise to cover unique use cases since the protocol is designed to be
-interface first. We imagine that businesses and marketing groups, for example, may desire
-to control a Social Identity using a different set of rules than the default
-implementation.
+interface first. We imagine that businesses and marketing groups, for example, may desire to
+control a Social Identity using a different set of rules than the default implementation.
```

Comment:
Well spotted, I removed similar language elsewhere.

Reply:
Well.. I left this in there to speak to the fact that there could be different implementations. The sense of it in terms of user choice has changed from "different smart contracts" to "different DSNP systems". Whether this would rise to the level of "a different set of rules" is debatable. How about changing "different set of rules than the default implementation" to "a different implementation strategy"?

Reply:
I'm not sure that captures the point unless DSNP systems are cross compatible, but not sure how that would work for something around user control of the logic